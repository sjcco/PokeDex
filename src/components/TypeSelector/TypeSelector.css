.container {
  display: flex;
  flex-direction: column;
  height: 70vh;
  width: 45%;
  position: fixed;
  top: 28px;
  right: -10px;
  border-bottom-left-radius: 0.5rem;
  box-shadow: 0.05rem 0.1rem 0.3rem -0.03rem rgba(0, 0, 0, 0.45);
  margin: 2rem 0.5rem;
  background-color: white;
  overflow-y: auto;
  overflow-x: hidden;
}

.container:last-child {
  margin-bottom: 2rem;
}

.btn {
  background-color: transparent;
  border: none;
  margin: 0.5rem 0;
}
