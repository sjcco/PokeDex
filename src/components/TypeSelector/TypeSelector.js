import React from 'react';
import TypeBadge from '../TypeBadge/TypeBadge';
import { container, btn } from './TypeSelector.css';

const TypeSelector = ({ filter }) => {
  const types = ['normal', 'fire', 'water', 'grass', 'flying', 'fighting', 'poison', 'electric', 'ground', 'rock', 'psychic', 'ice', 'bug', 'ghost', 'steel', 'dragon', 'dark', 'fairy'];
  return (
    <div className={container}>
      <button className={btn} onClick={() => (filter('all'))} type="button">All</button>
      {types.map(type => (
        <button className={btn} key={type} onClick={() => (filter(type))} type="button">
          <TypeBadge type={type} />
        </button>
      ))}
    </div>
  );
};

export default TypeSelector;
