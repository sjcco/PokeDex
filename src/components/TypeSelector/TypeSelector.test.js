/**
 * @jest-environment jsdom
 */

import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import TypeSelector from './TypeSelector';

test('renders content', () => {
  const fnc = jest.fn();
  const component = renderer.create(<TypeSelector
    filter={fnc}
  />).toJSON();
  expect(component).toMatchSnapshot();
});
