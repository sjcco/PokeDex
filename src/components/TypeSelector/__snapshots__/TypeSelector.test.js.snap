// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content 1`] = `
<div
  className="container"
>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    All
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <div
      className="pkmnType"
      style={
        Object {
          "backgroundColor": "#a8a878",
        }
      }
    >
      <span>
        normal
      </span>
    </div>
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <div
      className="pkmnType"
      style={
        Object {
          "backgroundColor": "#f08030",
        }
      }
    >
      <span>
        fire
      </span>
    </div>
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <div
      className="pkmnType"
      style={
        Object {
          "backgroundColor": "#6890f0",
        }
      }
    >
      <span>
        water
      </span>
    </div>
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <div
      className="pkmnType"
      style={
        Object {
          "backgroundColor": "#98d888",
        }
      }
    >
      <span>
        grass
      </span>
    </div>
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <div
      className="pkmnType"
      style={
        Object {
          "backgroundColor": "#a890f0",
        }
      }
    >
      <span>
        flying
      </span>
    </div>
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <div
      className="pkmnType"
      style={
        Object {
          "backgroundColor": "#c03028",
        }
      }
    >
      <span>
        fighting
      </span>
    </div>
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <div
      className="pkmnType"
      style={
        Object {
          "backgroundColor": "#a040a0",
        }
      }
    >
      <span>
        poison
      </span>
    </div>
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <div
      className="pkmnType"
      style={
        Object {
          "backgroundColor": "#f8d030",
        }
      }
    >
      <span>
        electric
      </span>
    </div>
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <div
      className="pkmnType"
      style={
        Object {
          "backgroundColor": "#e0c068",
        }
      }
    >
      <span>
        ground
      </span>
    </div>
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <div
      className="pkmnType"
      style={
        Object {
          "backgroundColor": "#b8a038",
        }
      }
    >
      <span>
        rock
      </span>
    </div>
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <div
      className="pkmnType"
      style={
        Object {
          "backgroundColor": "#f85888",
        }
      }
    >
      <span>
        psychic
      </span>
    </div>
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <div
      className="pkmnType"
      style={
        Object {
          "backgroundColor": "#98d8d8",
        }
      }
    >
      <span>
        ice
      </span>
    </div>
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <div
      className="pkmnType"
      style={
        Object {
          "backgroundColor": "#a8b820",
        }
      }
    >
      <span>
        bug
      </span>
    </div>
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <div
      className="pkmnType"
      style={
        Object {
          "backgroundColor": "#705898",
        }
      }
    >
      <span>
        ghost
      </span>
    </div>
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <div
      className="pkmnType"
      style={
        Object {
          "backgroundColor": "#b8b8d0",
        }
      }
    >
      <span>
        steel
      </span>
    </div>
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <div
      className="pkmnType"
      style={
        Object {
          "backgroundColor": "#7038f8",
        }
      }
    >
      <span>
        dragon
      </span>
    </div>
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <div
      className="pkmnType"
      style={
        Object {
          "backgroundColor": "#705848",
        }
      }
    >
      <span>
        dark
      </span>
    </div>
  </button>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <div
      className="pkmnType"
      style={
        Object {
          "backgroundColor": "#f0b6bc",
        }
      }
    >
      <span>
        fairy
      </span>
    </div>
  </button>
</div>
`;
